source/extensions/filters/http/aws_lambda/aws_lambda_filter.cc

```diff
@@ -54,13 +65,48 @@ bool isTargetClusterLambdaGateway(Upstream::ClusterInfo const& cluster_info) {
   return egress_gateway_it->second.bool_value();
 }
 
+bool isContentTypeTextual(const Http::RequestOrResponseHeaderMap& headers) {
+  // If transfer-encoding is anything other than 'identity' (i.e. chunked, compress, deflate or
+  // gzip) then we want to base64-encode the response body regardless of the content-type value.
+  if (auto encoding_header = headers.TransferEncoding()) {
+    if (!absl::EqualsIgnoreCase(encoding_header->value().getStringView(), "identity")) {
+      return false;
+    }
+  }
+
+  // If we don't know the content-type, then we can't make any assumptions.
+  if (!headers.ContentType()) {
+    return false;
+  }
+
+  const Http::LowerCaseString content_type_value{
+      std::string(headers.ContentType()->value().getStringView())};
+  if (content_type_value.get() == "application/json") {
```

Comment:
Some of these content_types are already in headers.h, please use them if you can